src/olympia/blocklist/models.py

```diff
@@ -150,6 +149,36 @@ def review_unlisted_link(self):
     def active_submissions(self):
         return BlocklistSubmission.get_submissions_from_guid(self.guid)
 
+    @classmethod
+    def get_blocks_from_guids(cls, guids):
+        """Given a list of guids, return a list of Blocks - either existing
+        instances if the guid exists in a Block, or new instances otherwise.
+        """
+        # load all the Addon instances together
+        addons = list(Addon.unfiltered.filter(
+            guid__in=guids).no_transforms())
```

Comment:
Probably want to force the use of the replica here and in the `existing_blocks` fetching below, for a bulk operation it becomes worth it I think.

Reply:
This is probably already happening in the management command case though, since there is no `use_primary_db` decorator or things like that, so carry on :)